File: 182737/java/StressTest.java
import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;
import java.net.*;

public class StressTest {
	static boolean isAlive = true;
	static String host = "localhost";
	static int port = 34000;
	public static void main(String[] args) {
		if (args.length > 0)
			host = args[0];
		if (args.length > 1)
			port = Integer.parseInt(args[1]);
		try {
			class vlakno_pozadavku extends Thread {
				public void run() {
                    Socket s = null;

					try {
						s = new Socket(host, port);
						InputStream is = s.getInputStream();
						OutputStream os = s.getOutputStream();
						long time = System.nanoTime();
						byte request[] = ("GET / HTTP/1.1\r\n"
								+ "Keep-Alive: timeout=5, max=1000\r\n\r\n").getBytes();
						byte pole[] = new byte[10000];
						int bytesIn = 0;
						int bytesOut = 0;
						while (isAlive) {
							os.write(request);
							os.flush();
							bytesOut += request.length;
							int status = is.read(pole);
							if (status == -1){
                                s.close();
                                isAlive = false;
                                break;
            
								
                            } else {
								bytesIn += status;
                            }
							long currentTime = System.nanoTime();
							if (currentTime - time > 1000000000) {
								System.out.println("bytesIn (kB/s): " + bytesIn / 1024 + ", bytesOut (kB/s): " + bytesOut / 1024);
								time = currentTime;
								bytesIn = 0;
								bytesOut = 0;
							}
						}

                        s.close();
						System.out.println("End of thread.");
					} catch (Exception e) {
						e.printStackTrace();

                        

                            try {
                                if (s != null){
                                s.close();
                                }
                            } catch(Exception ex){

                            }
                       
                        
                       
					}
				}
			}
			
			for (int i = 0; i < 10000; i++) {
				vlakno_pozadavku v = new vlakno_pozadavku();
				v.start();
			}
			
			int read;
			System.out.println("ENTER = stop");
			System.in.read();
			System.out.println("Connection halted.");
			isAlive = false;
			System.exit(0);
		} catch (Exception e) {
			e.printStackTrace();
		}
	}
}

File: StressTest.js
import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;
import java.net.Socket;

 class StressTest {
    static boolean isAlive = true;

    public static void main(String[] args){
        try {


            String host = "www.idnes.cz";
            int port = 80;

            if (args.length > 0){
                host = args[0];
                if (args.length > 1){
                    port = Integer.parseInt(args[1]);
                }
            }



            Socket socket = new Socket(host, port);
            InputStream input = socket.getInputStream();
            OutputStream output = socket.getOutputStream();

             class MyThread extends Thread {
                public void run(){
                    try {

                       // RandomStringGenerator pwdGenerator = new RandomStringGenerator.Builder().withinRange(33, 45)
                               // .build();
                      //  return pwdGenerator.generate(length);

                        while (isAlive) {
                            output.write("GET / HTTP/1.1\r\n Keep alive: timeout=5 max=10000\r\n\r\n".getBytes());
                            output.flush();
                            byte array[] = new byte[1000];
                            int status = input.read(array);
                            if (status == -1){
                                System.out.println("End of the thread.");
                            }
                        }

                    }  catch (IOException e){
                        e.printStackTrace();
                    }

                       /* } catch (IOException e){
                            e.printStackTrace();
                        }*/
                }
            }


            for (int i = 0; i < 10000; i++){
                MyThread v = new MyThread();
                v.start();
            }


            int read;
            System.out.println("konec = enter");
            System.in.read();
            System.out.println("Nashle");
            isAlive = false;   // not needed, we cannot renew sleep caused by OS, only way out is keyboard press
            System.exit(0);
        } catch (Exception e){
            e.printStackTrace();
        }




    }
}
